### PR TITLE
Add setter/getter for Pose's each element - Issue 35

### PR DESCRIPTION
**Pull request by pxalcantara**
Solved #35 

Add methods to Pose3.hh to get X,Y,Z,Roll,Pitch and Yaw directly, as suggested in the issue description : 

- pose.Pos.X() -> pose.X()
- pose.Pos.Y() -> pose.Y()
- pose.Pos.Z() -> pose.Z()
- pose.Rot.Roll() -> pose.Roll()
- pose.Rot.Pitch() -> pose.Pitch()
- pose.Rot.Yaw() -> pose.Yaw()

### CODE REVIEW

**Comment by pxalcantara** [comment]
@chapulina Here we go! first PR!! \o/

File: src/Pose_TEST.cc [hunk 1]
```diff
@@ -155,3 +155,22 @@ TEST(PoseTest, OperatorStreamOut)
...
+TEST(PoseTest, PoseElements)
+{
+  const math::Pose3i pose(0, 1, 2, 0, 0, 0);
+
+  int x = pose.Pos().X();
+  int y = pose.Pos().Y();
+  int z = pose.Pos().Z();
+  int Roll = pose.Rot().Roll();
+  int Pitch = pose.Rot().Pitch();
+  int Yaw = pose.Rot().Yaw();
```
Comment (chapulina) [inline src/Pose_TEST.cc:168]: Small style detail: could you use lowercase for all variables? Such as:

```suggestion
  int yaw = pose.Rot().Yaw();
```

Comment (pxalcantara) [reply src/Pose_TEST.cc:168]: ohh yeah, sure, I can do this. I've forget about the variable style code, sorry!

**Comment by scpeters** [comment]
are you able to merge this branch with `ign-math6`? it is currently out of date with the base branch

**Comment by pxalcantara** [comment]
[DO NOT MERGE] @chapulina @scpeters 
This code is not finished. The approach used to implement the mutable X(), Y() and Z() can not be used in Roll(), Pitch() and Yaw(). When this approach is used I got these errors:

RPY_error

That is expected because the Quaternion.Roll() , Pitch() and Yaw() are `const` : https://github.com/ignitionrobotics/ign-math/blob/ca49e983bc3939b6626e78f036ff5ea9bf10d270/include/ignition/math/Quaternion.hh#L423

How should I solve this point? I thought in some ways but, all of them I need to edit the library Quaternion.hh and, also, there is pattern and architecture questions, so I decided to ask

**Comment by chapulina** [comment]
humm nice catch

In fact, this reminds me that we've been deprecating the mutable accessors in favor of `Set*` functions, see the [Migration guide](https://github.com/ignitionrobotics/ign-math/blob/master/Migration.md) and issue #101 .

So I think the solution is to substitute the mutable functions with `Set` versions.

**Comment by pxalcantara** [comment]
Humm, nice, so, maybe, stop working with this issue and help with this functionality of the Quaternio.hh ? After that continue to working here?

**Comment by chapulina** [comment]
Oh I don't think you need to address the entire issue. I'd just change the new functions you're adding. Remove all the mutable functions and add:

```
Pose3d::SetX
Pose3d::SetY
Pose3d::SetZ
Pose3d::SetRoll
Pose3d::SetPitch
Pose3d::SetYaw
```

**Comment by pxalcantara** [comment]
Ok, good, should I remove Pos() and Rot() mutable, as well, and change for SetPos() and SetRot(), or is it better to do this in another PR?

**Comment by chapulina** [comment]
> should I remove Pos() and Rot() mutable, as well, and change for SetPos() and SetRot()

Good call, yes, please!

**Comment by scpeters** [comment]
> Ok, good, should I remove Pos() and Rot() mutable, as well, and change for SetPos() and SetRot(), or is it better to do this in another PR?

those API's are already released in ign-math6, so I wouldn't remove them in this PR

**Comment by chapulina** [comment]
> those API's are already released in ign-math6, so I wouldn't remove them in this PR

Ah sorry, thank you Steve. Let's only remove the ones being added by this PR.

**Comment by scpeters** [comment]
> This code is not finished. The approach used to implement the mutable X(), Y() and Z() can not be used in Roll(), Pitch() and Yaw().

Quaternion doesn't allow modification of Euler angles independently; it only allows them to all be set at once. I would suggest simply adding `Get` functions for roll, pitch, yaw in this PR and address setting them in a separate pull request

**Comment by pxalcantara** [comment]
> > This code is not finished. The approach used to implement the mutable X(), Y() and Z() can not be used in Roll(), Pitch() and Yaw().
> 
> Quaternion doesn't allow modification of Euler angles independently; it only allows them to all be set at once. I would suggest simply adding `Get` functions for roll, pitch, yaw in this PR and address setting them in a separate pull request

Ok, It was I thought, and also using the Euler method would not be a good idea now since it is deprecated, so I'll do your suggestion!
Thanks @scpeters and @chapulina

**Comment by pxalcantara** [comment]
@scpeters @chapulina sorry for the delay. Is this the way that you suggested? About the failure of the CI test, I couldn't understand the error, what should I do to fix it?

**Comment by scpeters** [comment]
it looks like the build machines had an error; I'll start the jobs again

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Comment by pxalcantara** [comment]
Thank you @scpeters!

**Pull request merged**